compiler/src/iree/compiler/Codegen/Common/GPU/GPUHeuristics.cpp

```diff
@@ -690,15 +692,24 @@ FailureOr<std::pair<GPUMMASchedule, GPUMMASchedule>> deduceAttentionSchedule(
                                     canUpcastAcc, mustBeAligned))) {
         continue;
       }
-      // Check if we can reuse the output of intrinsicA for lhs/rhs of
-      // intrinsicB.
+
       auto matchLayout =
           [](IREE::GPU::MMASingleSubgroupLayout layoutA,
              IREE::GPU::MMASingleSubgroupLayout layoutB) -> bool {
         return (layoutA.element == layoutB.element) &&
                (layoutA.thread == layoutB.thread) &&
                (layoutA.tstrides == layoutB.tstrides);
       };
+
+      if (!matchLayout(getSingleSubgroupLayout(intrinsicA.mmaKind,
+                                               IREE::GPU::MMAFragment::Acc),
+                       getSingleSubgroupLayout(intrinsicB.mmaKind,
+                                               IREE::GPU::MMAFragment::Acc))) {
+        continue;
+      }
```

Comment:
Nice, I think we can improve the matchLayout check, but thats for a later pr. This  is good enough for now.